gnosis/safe-react-apps
# [TX Builder] Separate address field in address and ABI

## What it solves
Resolves #305 

## How this PR fixes it
We are adding a new JSON Field for displaying and managing the Contract ABI interaction. Now the address and the ABI are separated.

## How to test it
1) Enter valid address with retrieved ABI. JSON field should be filled in after the correct retrieval and the methods interaction showed up
2) Enter valid address with no ABI retrieval should show only the ETH send 
3) You can edit the ABI manually
4) If you try to paste in the ABI field when it has some value a popup should be showed asking if the ABI replacement is intended

**TX Builder should work same way adding and sending transactions**

## Screenshots

![image](https://user-images.githubusercontent.com/1576776/152964971-96351157-2b18-4dcf-ac25-2eb421574fb2.png)

![image](https://user-images.githubusercontent.com/1576776/152965224-a676af92-641a-4a3a-af9b-85f09182c80d.png)

![image](https://user-images.githubusercontent.com/1576776/152965323-224fbe68-6733-4069-b260-612c6a81224d.png)
